### DISCUSSION TITLE
Improve input latency in single threaded mode

### DISCUSSION THREAD

**Discussion by Modsnotworking** [ppy/osu#22420] [Q&A] [answered]
I am experiencing really bad in put latency in single threaded mode. Also the game used to run really greatly before in multi threaded mode but now it just lags[in the gameplay part] but I get less input latency. 

![osu_2023-01-25_21-51-27](https://user-images.githubusercontent.com/70589304/214619695-a939321d-8de7-4d84-b1ed-5dd7e43903c6.jpg)

tl;dr please improve the input latency in single threaded mode

**Comment by Theighlin** [comment]
That is not input latency, it's frame time. At best i would call it input polling, that is: your inputs are only checked once every 33 milliseconds, but they shuold not be late, when this check happens. This is due to running in singlethread, because the input thread which is very fast, has to wait for the draw thread (the fps you see there) to process a frame.

The simple answer is, optimisations are a work in progress, and the game will run better in time.

**Comment by peppy** [comment, marked as answer]
I'm very surprised it's running *worse* than previously. That might be due to external changes, like GPU driver updates. We've only been making improvements from our end in terms of performance. Hard to really say anything as you haven't specified any specs or platform.

That said, we have a new rendering engine coming online soon which might help, so keep checking back as new releases drop.
